zetaclient/zetacore_observer.go

```diff
@@ -155,17 +158,29 @@ func (co *CoreObserver) startSendScheduler() {
 							co.logger.ZetaChainWatcher.Error().Err(err).Msgf("failed to get prometheus gauge: %s", metrics.PendingTxs)
 							continue
 						}
-						gauge.Set(float64(len(sendList)))
+						gauge.Set(float64(len(cctxList)))
 
-						for idx, send := range sendList {
-							params := send.GetCurrentOutTxParam()
+						for idx, cctx := range cctxList {
+							params := cctx.GetCurrentOutTxParam()
 							if params.ReceiverChainId != c.ChainId {
-								log.Warn().Msgf("mismatch chainid: want %d, got %d", c.ChainId, params.ReceiverChainId)
+								co.logger.ZetaChainWatcher.Error().Msgf("mismatch chainid: want %d, got %d", c.ChainId, params.ReceiverChainId)
+								continue
+							}
+							// #nosec G701 range is verified
+							currentHeight := uint64(bn)
+							nonce := params.OutboundTxTssNonce
+							outTxID := fmt.Sprintf("%s-%d-%d", cctx.Index, params.ReceiverChainId, nonce) // would outTxID a better ID?
+
+							// Process Bitcoin OutTx
+							if common.IsBitcoinChain(c.ChainId) && !outTxMan.IsOutTxActive(outTxID) {
```

Comment:
Are you processing and retrying bitcoin outbound txns every iteration? Will this cause a lot of unnecessary tss signing?

Reply:
It's a good point. It schedules at most 1 keysign per Zeta block now to target TPM of 5. This part still has room to be optimized in some way. (e.g., if we already have a broadcasted outTx, not included yet, we should wait?)

Reply:
@kevinssgh [cca3a](https://github.com/zeta-chain/node/pull/1147/commits/cca3a22d92c7b7a2ccbb6556d77ceacbdf398d39)